### ISSUE TITLE
PocoCpp error when building with MinGW

### ISSUE THREAD

**Issue by steinio** [cpp-pm/hunter#272]
Adding pococpp as specified in the example, using the latest hunter tag with cmake version 3.17.3 that came with MSYS2, compiling with mingw gcc version 10.1.0.

Error is `C:/Users/steinito/.hunter/_Base/43ffd06/a2d374d/647ab83/Build/PocoCpp/Source/Net/include/Poco/Net/ICMPv4PacketImpl.h:72:3: error: expected identifier before '(' token
   72 |   TIMESTAMP_REQUEST,`

I can build the same project on ubuntu without any errors, so the project and CMakeLists.txt do not have any errors.
I can also build it with Visual Studio 2017 using cmake as well, with no errors.

I have noticed someone posting duplicate error for Conan, so possibly the issue is with either mingw or poco?

**Comment by rahulshethsc** [comment]
Yeah that sounds right. We can patch the sources if you have a workaround.

**Comment by steinio** [comment]
Issue comes from a [mingw definition](https://github.com/pocoproject/poco/issues/3121) that matches the above mentioned PocoCpp definition.

I am currenty only using NET, FOUNDATION, UTIL and JSON, and I am able to build pococpp as an external library, but not from hunter, because it tries to build everything.

I am pretty new to hunter, but tried to define only the packages I needed in a config.cmake.
I put this file in cmake/Hunter/config.cmake, and contains

```
hunter_config(PocoCpp
VERSION 1.10.0
CMAKE_ARGS
ENABLE_DATA=OFF
ENABLE_NETSSL=OFF
ENABLE_CRYPTO=OFF
ENABLE_DATA=OFF
ENABLE_DATA=OFF
ENABLE_DATA=OFF
ENABLE_DATA=OFF
ENABLE_DATA_ODBC=OFF
ENABLE_MONGODB=OFF
ENABLE_REDIS=OFF
ENABLE_PDF=OFF
ENABLE_APACHECONNECTOR=OFF
ENABLE_NETSSL_WIN=OFF
)
```

I then added `LOCAL` to my HunterGate file like so

```
HunterGate(
        URL "https://github.com/cpp-pm/hunter/archive/v0.23.273.tar.gz"
        SHA1 "43ffd06cf0352babdbe28d333adb15379ab59681"
	LOCAL
)
```

, but when building I get

`Version not found: 1.10.0. Please check 'hunter_config' command.`

This is the version used in the PocoCpp example, so figured it would work.
I have tried with earlier version too like 1.9.4, but I get the same bug.
Am I using the config.cmake wrong?

**Comment by NeroBurner** [comment]
try with the following, to configure the current default (in Hunter) version

```cmake
hunter_config(PocoCpp
  VERSION ${HUNTER_PocoCpp_VERSION} CMAKE_ARGS
  ENABLE_DATA=OFF
  ENABLE_NETSSL=OFF
  ENABLE_CRYPTO=OFF
  ENABLE_DATA=OFF
  ENABLE_DATA=OFF
  ENABLE_DATA=OFF
  ENABLE_DATA=OFF
  ENABLE_DATA_ODBC=OFF
  ENABLE_MONGODB=OFF
  ENABLE_REDIS=OFF
  ENABLE_PDF=OFF
  ENABLE_APACHECONNECTOR=OFF
  ENABLE_NETSSL_WIN=OFF
)
```
